core/src/keys.rs

```diff
@@ -26,3 +26,24 @@ pub trait Ownable {
     /// Returns the associated `StealthAddress`
     fn stealth_address(&self) -> stealth::StealthAddress;
 }
+
+pub(crate) fn owns_unchecked(
+    a: &JubJubScalar,
+    owner: &impl crate::Ownable,
+) -> bool {
+    let sa = owner.sync_address();
+    let aR = sa.R() * a;
+
+    sa.k() == &aR
+}
+
+/// This trait grants the ability to check if a note is owned.
+pub trait Ownability {
+    /// Checks if:
+    /// `note_pk ?= H(R · a) · G + B` for `ViewKey`
+    /// `note_pk ?= (H(R · a) + b) · G` for `SecretKey`
+    fn owns(&self, owner: &impl crate::Ownable) -> bool;
+
+    /// Checks if `k_sync ?= R_sync · a`
+    fn owns_unchecked(&self, owner: &impl crate::Ownable) -> bool;
+}
```

Comment:
I think you can get away with only requiring a `fn a` function that returns the internal `a`.
`owns` and `owns_unchecked` can by computed from that, which means that the type that implements the `Ownability` trait doesn't need to re-compute the checks themselves.

```suggestion
pub trait Ownability {
    /// Checks if:
    /// `note_pk ?= H(R · a) · G + B` for `ViewKey`
    /// `note_pk ?= (H(R · a) + b) · G` for `SecretKey`
    fn owns(&self, owner: &impl crate::Ownable) -> bool;

    /// Checks if `k_sync ?= R_sync · a`
    fn owns_unchecked(&self, owner: &impl crate::Ownable) -> bool;
}
```

Reply:
With the `Ownability` trait we should also let the function `note.decrypt_data` (and with that also `note.value` and note.blinding_factor`) to take any type that implements `Ownability`.
This means that we can also decrypt the value and blinding factor with a `SecretKey`

Reply:
@moCello regarding your first comment: but then, how am I supposed to get `b` or `B`? In the end I still need two different ways to perform the check.

Regarding the decrypt data thing, I can look into it in a new commit.